docs/manual/config/keys.rst

```diff
@@ -45,6 +45,24 @@ The :class:`EzKey` modifier keys (i.e. ``MASC``) can be overwritten through the
        'C': 'control',
     }
 
+Callbacks can also be configured to work only under certain conditions by using
+the ``when()`` method. Currently, two conditions are supported:
+
+::  
+
+    from libqtile.config import Key
+
+    def myfunc():
+        pass
+
+    keys = [
+        # Only trigger callback for a specific layout
+        Key([mod], "a", lazy.function(myfunc).when(layout='max'))
```

Comment:
Maybe for the layout example we could put two commands into the `Key`, with different layouts, e.g.:
```py
Key(
    [mod, 'shift'],
    "j",
    lazy.layout.grow().when(layout='verticaltile'),
    lazy.layout.grow_down().when(layout='columns'),
)
```
This seems like it would be the main use-case.

Reply:
Sounds good. Would still be good to include an example for the `when_floating` condition.

Reply:
> Would still be good to include an example for the when_floating condition.

for sure

> Not sure what use case was envisaged for that.

Yeah me neither - guess it doesn't really matter then what the lazy thing is